DOCS-5856 integrated Community and Exchange search (#108)
mulesoft/docs-api-community-manager: modules/ROOT/pages/acm-lightning-components.adoc

Many ACM Lightning components show data from Anypoint Platform. ACM admins can change some of these data fields, such as API names, descriptions, and icons. ACM saves these changes but does not push the changes back to Anypoint Platform.

== ACM Search Box [[acm-search-box]]

> The ACM Search Box component enables users to search both for content that exists in Community Cloud, such as knowledge articles, support cases, and announcements, and content that exists in Anypoint Platform, such as APIs and client applications. Results from the search are presented in the <<acm-search-results>> component.

Reviewer: ```suggestion
The Anypoint API Community Manager Search Box component enables users to search for both content that exists in Community Cloud (such as knowledge articles, support cases, and announcements) and content that exists in Anypoint Platform (such as APIs and client applications). Results from the search are presented in the <<acm-search-results>> component.
```

> You can specify values for the following parameters of the ACM Search Box:

Reviewer: ```suggestion
You can specify values for the following parameters of the API Community Manager Search Box:
```

Component Title:: Specifies the title for the search input field.
Number of Results - Instant Incremental Response:: Specifies the number of results shown as the user types a query.
Search Areas:: Selects which types of content to search. Choices include APIs, Applications, Knowledge Articles, Cases, Pages, and Forums.
Linked Result Page:: Specifies the URL of the search results page shown after the user enters a search.

> === Search Configuration

Reviewer: Is this solitary subtopic necessary, as a subtopic?
You typically want at least two before you break a topic into subs....
Perhaps the container topic should have a title that incorporates this info?
Or...is this actually a mislabeled Level-2 peer topic? These are all components?

Author: Good point. The only two components are "Search Box" and "Search Results", and this configuration info applies to the "Search Box", so it's better without a separate heading. Removed it.

> Search is a basic expected feature in any developer portal, and becomes important as the number of APIs grows to around 50 and beyond.

Reviewer: This intro sentence sounds gratuitous to me. We all know about Search and why it is important for large amounts of data....

Author: True, better to omit this. Removed.

> The example portals and default pages have one ACM Search Box and one ACM Search Results component. This is the best practice to minimize the chance of misconfigurations.

Reviewer: The example portals and default pages of what? Some kind of template?
Is this info necessary here?
Why would there be more than one search or results component?

Author: API Community Manager comes with a default setup, and community admins can add, remove, and move pages and components to rebuild the community site in any way. Admins may wish to separate site members into audiences, such as internal developers at their organization and external developers at other organizations, and each audience could have their own search, which could automatically filter out APIs that would only be used by the other group. However, in most cases a community will just have one search component and one results component. I've rewritten this part for better clarity.

> The ACM Search Box parameter Linked Result Page specifies the search results page. That page must have an ACM Search Results component to display the search results.

Reviewer: Should this sentence be up above, where the parameters are described?
And...can you reread these two sentences? I can't make any sense of them, but maybe I am having some sort of block....

Author: Yes, better to move those up. I also rewrote for clarity: "The Linked Result Page parameter specifies the URL of a search results page, and the search results page must include an ACM Search Results component to display the search results. If the page does not include an ACM Search Results component, members and guest users who enter a search will go to the page and see no results."

> The ACM Search Results component shows only results of the content types specified in the Search Areas parameters of the ACM Search Box component. If a content type is not selected then the search results do not include any results of that content type.

Reviewer: Shouldn't this text be moved to the following subtopic by the same name?
Also, GLOBALLY fully spell or use the accepted short form API Community Manager rather than ACM.
```suggestion
The API Community Manager Search Results component shows results of only the content types specified in the Search Areas parameters of the API Community Manager Search Box component. If a content type is not selected then the search results do not include any results of that content type.
```

Author: Rewrote this to clarify that it's focused on the Search Areas parameter.

Unfortunately the UI has a long alphabetical list of components, including the names "ACM Search Box" and "ACM Search Results", so if the docs call them "API Community Manager Search Box" and so on, we'll send the users to the wrong place in the list. I'll raise this with the PM next week when I scrub "ACM" from all pages in the docs, and hopefully we can get it fixed in the UI.

> IMPORTANT: The current version of the integrated search does not include CMS articles in the search results.

Reviewer: Can we specify the version number that we mean by "current," to insure against mistakes during updates?

Author: Yes, that's much better. Rewrote and will add to the final release to-do list to add the package version here.

> Search results are sorted by the date they were most recently modified, from the most recent date to the oldest.

Reviewer: ```suggestion
Search results are sorted by the date they were most recently modified, from newest to oldest.
```

> To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[Preview Your API Community] and run a search.

Reviewer: ```suggestion
To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[preview your API community] and run a search.
```

== ACM Search Results [[acm-search-results]]

> The ACM Search Results component shows results after a user enters a search in the <<acm-search-box>> component.

Reviewer: ```suggestion
The API Community Manager Search Results component shows results after a user enters a search in the <<acm-search-box>> component.
```

> You can specify values for the following parameters of the ACM Search Results:

Reviewer: ```suggestion
You can specify values for the following parameters of the API Community Manager Search Results:
```
OR
```suggestion
You can specify values for the following parameters of the Search Results:
```

Card Width:: Specifies the card width.
Card Height:: Specifies the card height.

== ACM Search Box

> The ACM Search Box component enables users to search data integrated from both Community Cloud and Anypoint Platform.

Reviewer 2: ... enables users to search for content that exists in both Community Cloud, like articles, announcements and forum posts, and the Anypoint Platform, like APIs and client applications. Results from the search are presented in the ACM Search Results component [ADD LINK]

Author: Made this change. Much better! Thank you.

Number of Results - Instant Incremental Response:: Specifies the number of results shown as the user types a query.
Search Areas:: Selects which types of content to search. Choices include APIs, Applications, Knowledge Articles, Cases, Pages, and Forums.
Linked Result Page:: Specifies the URL of the search results page shown after the user enters a search.

=== Search Configuration

Search is a basic expected feature in any developer portal, and becomes important as the number of APIs grows to around 50 and beyond.

The example portals and defaults have one Integrated Search Input and one Integrated Search Response Page. This is the best practice to minimize the chance of misconfigurations.

The Integrated Search Input parameter Linked Result Page specifies the search results page. That page must have an Integrated Search Response Page component to display the search results.

The Integrated Search Response Page component shows only results of the content types specified in the Search Areas parameters of the Integrated Search Input component. If a content type is not selected then the search results page does not display any search results of that content type.

> The current version of the integrated search does not include CMS articles in the search results.

Reviewer 3: This needs to be called out with `IMPORTANT`.

Author: Done. Thank you!

Search results are sorted by the date they were most recently modified, from the most recent date to the oldest.

To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[Preview Your API Community] and run a search.

== ACM Search Results

> The ACM Search Results component shows results after a user enters a search.

Reviewer 2: ... a user enters a search in the ACM Search Box component. [ADD LINK]

Author: Made this change. Thanks!
